File: Nth Fibonacci Number - GFG/nth-fibonacci-number.cpp
//{ Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++
class Solution {
  public:
    int nthFibonacci(int n){
        // code here
        int n1 =0;
        int n2 =1;
        int mod = 1000000007;
    
        for(int i=2; i<=n; i++){
            int sum = n1+n2;
            n1 = n2;
            n2 = sum%mod;
        }
        return n2;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution ob;
        cout << ob.nthFibonacci(n) << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Nth Fibonacci Number - GFG/nth-fibonacci-number.java
//{ Driver Code Starts
//Initial Template for Java
import java.util.*;
import java.io.*;
class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int n = Integer.parseInt(read.readLine());
            Solution ob = new Solution();

            System.out.println(ob.nthFibonacci(n));
        }
    }
}
// } Driver Code Ends


//User function Template for Java

//User function Template for Java
class Solution {
    static int nthFibonacci(int n){
        // code here
        int n1 =0;
        int n2 =1;
        int mod = 1000000007;
    
        for(int i=2; i<=n; i++){
            int sum = n1+n2;
            n1 = n2;
            n2 = sum%mod;
        }
        return n2;
        
    }
}